File: hms-frontend/src/services/billingService.js
import api from "./api";

export const getAllBills = () => api.get("/billing");

export const addBill = (bill) => api.post("/billing", bill);

export const deleteBill = (id) => api.delete(`/billing/${id}`);

export const getBillById = (id) => api.get(`/billing/${id}`);

export const updateBill = (id, bill) =>
  api.put(`/billing/${id}`, bill);

File: hms-backend/src/main/java/com/hms/service/BillingService.java
package com.hms.service;

import java.util.List;
import com.hms.entity.Billing;

public interface BillingService {

    Billing saveBilling(Billing billing);

    List<Billing> getAllBills();

    Billing getBillById(Long id);

    Billing updateBilling(Long id, Billing billing);

    void deleteBill(Long id);
}